File: haskell/quicksort.hs
quicksort :: (Ord a) => [a] -> [a]  
quicksort [] = []  
quicksort (x:xs) =   
    let smaller = quicksort [a | a <- xs, a <= x]  
        bigger = quicksort [a | a <- xs, a > x]  
    in  smaller ++ [x] ++ bigger 


main = do 
    print(quicksort [7,15,12,-5,14,88,52,11])
    print(quicksort "Haskell")
    print(quicksort [1.5,9.85,9.99,-2.58])
	

File: php/quicksort.php
<?php

/**
 * Based on the haskell quicksort
 *
 * @param array $array
 *
 * @return array
 */
function quickSort(array $array) {

    // base case
    if (empty($array)) {
        return $array;
    }

    $head = array_shift($array);
    $tail = $array;

    $lesser = array_filter($tail, function ($item) use ($head) {
        return $item <= $head;
    });

    $bigger = array_filter($tail, function ($item) use ($head) {
        return $item > $head;
    });

    return array_merge(quickSort($lesser), [$head], quickSort($bigger));
}